ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -574,12 +574,13 @@ forwardTableKeySets dblog = \(UnforwardedReadSets seqNo' values keys) ->
       ApplyValuesMK $ forwardValuesAndKeys values keys (cumulativeDiffSeqUtxoDiff diffs)
 
 dbChangelogPrefix ::
-     ( HasHeader blk, HeaderHash blk ~ HeaderHash (l EmptyMK)
+     ( HeaderHash blk ~ HeaderHash (l EmptyMK)
      , GetTip (l EmptyMK)
      , TableStuff l
+     , StandardHash (l EmptyMK)
      )
   => Point blk -> DbChangelog l -> Maybe (DbChangelog l)
-dbChangelogPrefix = prefixDbChangelog
+dbChangelogPrefix pt = prefixDbChangelog (castPoint pt)
```

Comment:
For this small function you can probably write it point-free
```suggestion
dbChangelogPrefix = prefixDbChangelog . castPoint
```

Reply:
Also, having both `dbChangelogPrefix` and `prefixDbChangelog` is a bit confusing. Consider simply inlining `dbChangelogPrefix`:
```diff
diff --git a/ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs b/ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs
index 6c75243c6..820e89f94 100644
--- a/ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs
+++ b/ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs
@@ -573,15 +573,6 @@ forwardTableKeySets dblog = \(UnforwardedReadSets seqNo' values keys) ->
     forward (ApplyValuesMK values) (ApplyKeysMK keys) (ApplySeqDiffMK diffs) =
       ApplyValuesMK $ forwardValuesAndKeys values keys (cumulativeDiffSeqUtxoDiff diffs)
 
-dbChangelogPrefix ::
-     ( HeaderHash blk ~ HeaderHash (l EmptyMK)
-     , GetTip (l EmptyMK)
-     , TableStuff l
-     , StandardHash (l EmptyMK)
-     )
-  => Point blk -> DbChangelog l -> Maybe (DbChangelog l)
-dbChangelogPrefix pt = prefixDbChangelog (castPoint pt)
-
 -- | Isolates the prefix of the changelog that should be flushed
 --
 -- TODO take some argument to bound the size of the resulting prefix?
@@ -762,7 +753,7 @@ ledgerDbPrefix pt db
                             (pointSlot pt)
                             ((== pt) . castPoint . getTip . unCheckpoint . either id id)
                             =<< ledgerDbCheckpoints db
-        dblog' <- dbChangelogPrefix pt $ ledgerDbChangelog db
+        dblog' <- prefixDbChangelog (castPoint pt) $ ledgerDbChangelog db
         return $ LedgerDB
                   { ledgerDbCheckpoints = checkpoints'
                   , ledgerDbChangelog   = dblog'
```